### DISCUSSION TITLE
Add mask for instance segmentation

### DISCUSSION THREAD

**Discussion by arashazimi0032** [obss/sahi#900] [Q&A] [answered]
Hello, I saw in your code that in the yolo v8 model, on line 62, you save only the output boxes from the model, it would be great if you also save the masks and display them in the output.

https://github.com/obss/sahi/blob/0414e4777fb74c38dffa32b0dc1275714c1e1d51/sahi/models/yolov8.py#L62

prediction_result = [
            result.boxes.data[result.boxes.data[:, 4] >= self.confidence_threshold] for result in prediction_result
        ]

**Comment by sweetlhare** [comment]


**Reply by filipemendesdev** [reply]
Hey @sweetlhare I was trying to replicate your results on my colab notebook and I can't for some reason. Would you mind posting your whole code just for comparison? I need to implement SAHI + YoloV8, displaying the masks like you did. 

![255220227-d8a14235-da64-48b8-885f-256d641d7d76](https://github.com/obss/sahi/assets/105338201/6356a2cf-279d-488f-9c8d-118cccd3a867)

Thank you in advance :)

**Comment by fcakyon** [comment, marked as answer]
Segmentation mask support for yolov8, mmdetection, detectron2, torchvision models is live with the latest release! @FilipeMendesVMD:  https://github.com/obss/sahi/discussions/1051

There is also 5x memory and speed improvement when working with masks!

Check the demo notebook on how it works: https://github.com/obss/sahi/blob/main/demo/inference_for_yolov8.ipynb
